data/events/scripts/player.lua

```diff
@@ -200,6 +200,12 @@ function Player:onReportBug(message, position, category)
 end
 
 function Player:onTurn(direction)
+	if self:getGroup():getAccess() and self:getDirection() == direction then
```

Comment:
That is not a good approach. You're embedding extra functionality in something that was just not designed to that.
"onTurn" is intended to make a turn not a walk. It comes from the client call that is also designed for turn, not walk. You're embedding logic and creating complexity in a function that was not designed to that, without the proper walking validations what could lead to issues. 
You should aim a solution that provides a proper benefit, not that just "pack" everything you think that is necessary in a single function because you want it to do tons of things for different scenarios, that is not functional and, definitely, not scalable. It's hell to give maintenance.